### DISCUSSION TITLE
How to bypass immediate parent constructor when extending a class?

### DISCUSSION THREAD

**Discussion by jbruni** [scandipwa/scandipwa#4865] [Q&A]
When extending/overriding a container or component class, it is common to have this call to the parent's constructor:

```js
    __construct(props) {
        super.__construct(props);
```

I am facing a scenario where the parent `__construct` shall not run, because the extended class shall behave differently.

**BUT**, a call to the "grandparent's" (React's ` PureComponent`) `__construct` is necessary.

I would like to do something as `super.super.__construct(props);` - but `super.super` is not supported.

How can I extend a component, and bypass its `__construct`, but still call PureComponent's `__construct`?

@carinadues - mentioning you because it may get good and faster attention (sorry to disturb...)  Thanks!

**Comment by zans-laksa** [comment]
Why don't you just create a new class without extending the base class and make it do what you want?
Just copy-paste it and remove unnecessary behavior.
